scripts/globals/npc_util.lua

```diff
@@ -7,25 +7,34 @@ npcUtil = {};
 -- Picks a new position for an NPC and excluding the current position.
 -- INPUT: npc = npcID, position = 2D table with coords: index, {x, y, z}
 -- RETURN: table index
-function npcUtil.pickNewPosition(npc, positionTable)
-
+function npcUtil.pickNewPosition(npc, positionTable, allowCurrentPosition)
     local npc = GetNPCByID(npc)
     local positionIndex = 1 -- Default to position one in the table if it can't be found.
     local tableSize = 0
+    local allowCurrentPosition = allowCurrentPosition or false;
+
     for i, v in ipairs(positionTable) do   -- Looking for the current position
 
-        -- Finding by comparing the NPC's coords
-        if (math.floor(v[1]) == math.floor(npc:getXPos()) and math.floor(v[2]) == math.floor(npc:getYPos()) and math.floor(v[3]) == math.floor(npc:getZPos())) then
-            positionIndex = i; -- Found where the NPC is!
+        if (allowCurrentPosition == "false") then
```

Comment:
I don't really know why, but this worked while I tested and I just redid a printf-based test and it still works:
[30/Jun] [10:09:00][Status] do_init: loading zones[30/Jun] [10:09:01][LUA Script] allowCurrentPosition: true - newPosition: 3
[30/Jun] [10:09:00][Status] do_init: loading zones[30/Jun] [10:09:02][LUA Script] allowCurrentPosition: false - newPosition: 51
...
[30/Jun] [10:19:04][LUA Script] allowCurrentPosition: true - newPosition: 1
[30/Jun] [10:20:21][LUA Script] allowCurrentPosition: true - newPosition: 1
[30/Jun] [10:22:11][LUA Script] allowCurrentPosition: true - newPosition: 4
[30/Jun] [10:23:50][LUA Script] allowCurrentPosition: true - newPosition: 4
[30/Jun] [10:25:21][LUA Script] allowCurrentPosition: true - newPosition: 3
[30/Jun] [10:27:01][LUA Script] allowCurrentPosition: true - newPosition: 2
[30/Jun] [10:28:15][LUA Script] allowCurrentPosition: true - newPosition: 2
[30/Jun] [10:29:29][LUA Script] allowCurrentPosition: true - newPosition: 2
...